spec/index.bs

```diff
@@ -6307,6 +6313,22 @@ enum GPULoadOp {
 };
 </script>
 
+<dl dfn-type=enum-value dfn-for=GPULoadOp>
+    : <dfn>"load"</dfn>
+    ::
+        Loads the existing value for this attachment into the render pass.
+
+        Note:
+        On some GPU hardware (primarily mobile), providing a clear-value is significantly cheaper
+        because it avoids loading data from main memory into tile-local memory.
+        On other GPU hardware (primarily desktop/laptop), providing {{GPULoadOp/"load"}} may be
```

Comment:
Can we get the information backing this statement somewhere visible? It was only mentioned by @Kangz IIRC. I'd like to get a better understanding of what's going on there before baking the conclusions into the spec.

Reply:
@bjjones do you have a link to the performance analysis you did of Babylon.js's mipmap generation code? I think that's where we found this behavior.

I think it can be explained by the fact that clearing textures is a separate operation on desktop GPUs (flipping the clear color compression bit and writing the clear color compression plane) that's both extra work and causes additional synchronization to happen. When using `load`, the ROP wouldn't load framebuffer data if it doesn't need to (for example the blend mode just writes the shader's output to memory).

Reply:
I don't have the data saved, but I'll try to reproduce the data and can provide a brief writeup later today or Monday.

Previously, in the [bug](https://bugs.chromium.org/p/dawn/issues/detail?id=587) where this issue was discussed, I wrote this:

> I also did further investigation into loadOp clear vs. load on Nvidia/Intel. I found that on Intel graphics, performance was improved by 3x when using loadOp load instead of loadOp clear. On Nvidia, I found that performance was the same for both loadOp values.

Regarding mobile architectures - I do believe the above statements are correct (i.e. clear is faster than load). I have no immediate ability to test myself, but [this article](https://community.arm.com/developer/tools-software/graphics/b/blog/posts/picking-the-most-efficient-load-store-operations) explains the mechanism behind it.

Reply:
Thanks for the link!

Reply:
Something must have been terribly wrong there if you had 3x performance slowdown from loadOp. Even the explanation here doesn't match up: if there is an extra write to each pixel (the worst case, possibly for Intel), then it should have been at most 2x slow.

Reply:
I just now ran a quick and similar experiment. I modified Dawn's CppHelloTriangle example to use the different loadOps and took PIX frame captures. This test was done on Windows, with an Intel 630 and an Nvidia 3090. Below is the approximate durations of the render pass as reported by PIX:

|  Intel, LoadOp::Load | Intel, LoadOp::Clear  | 
|---|---|
|  32,000 ns | 76,000 ns  | 

|  Nvidia, LoadOp::Load | Nvidia, LoadOp::Clear  | 
|---|---|
|  5,100 ns | 5,100 ns  |  

The takeaways from my previous statement remain valid from this experiment:
- LoadOp::Load is faster than LoadOp::Clear on my Intel integrated GPUs
- LoadOp::Load and LoadOp::Clear performance costs are the same on my Nvidia 3090

Reply:
What if you do a full-screen quad that just clears the target before the draw call (that does the mipmap generation)? Would it be faster than `LoadOp::Clear`?

Reply:
I've been experimenting more and it seems my data on Nvidia was inadequate - the previous tests have been fairly small textures (above data had a 640x480 render target). When I run the same experiment with an increased render target size at 3840x2160, I do see that clear takes much longer than load - which is more consistent with the Intel data.

| Nvidia, 4k RT, LoadOp::Load  | Nvidia, 4k RT, LoadOp::Clear | 
|---|---|
| 12,000 ns |  41,000 ns | 

I am going to reach out to our driver team to see if I can get some comments on the numbers I'm seeing.

> What if you do a full-screen quad that just clears the target before the draw call (that does the mipmap generation)? Would it be faster than LoadOp::Clear?

If I understood correctly - this takes longer. Just directly drawing a quad with `FragColor = vec4<f32>(0.0,0.0,0.0,1.0)`; using DrawIndexedInstanced required 133,000ns on a 640x480 texture. The equivalent ClearRenderTargetView operation is taking ~45,000ns.